core/src/main/scala/org/apache/spark/scheduler/ExecutorLossReason.scala

```diff
@@ -58,3 +58,11 @@ private [spark] object LossReasonPending extends ExecutorLossReason("Pending los
 private[spark]
 case class SlaveLost(_message: String = "Slave lost", workerLost: Boolean = false)
   extends ExecutorLossReason(_message)
+
+/**
+ * A loss reason that means the worker is marked for decommissioning.
+ *
+ * This is used by the task scheduler to remove state associated with the executor, but
+ * not yet fail any tasks that were running in the executor before the executor is "fully" lost.
+ */
+private [spark] object WorkerDecommission extends ExecutorLossReason("Worker Decommission.")
```

Comment:
As with my comments in your spec I really dislike using the term `Worker` outside of the standalone implementation.

If you're talking about the executor being decommissioned (regardless of the underlying reason), it should be `ExecutorDecommissioned`.

If this is more specific (i.e. it really means the standalone worker, or the host where the executor is running), I'd call this "HostDecommission", and then it seems you're missing a separate loss reason for the executor being preempted.